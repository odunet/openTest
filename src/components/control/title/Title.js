import React, { useState, useContext } from 'react';
import { TextLabel, FlexRowDiv } from '../../../shared/styledComponents';
import { TextInput } from './Title.style';
import { changeTitle_Action } from '../../../actions';
import { store } from '../../../store';

export default function Title() {
  const [title, setTitle] = useState('');

  // Bring in dispatch
  const { dispatch } = useContext(store);

  const handleInput = ({ target }) => {
    setTitle(target.value);
    dispatch(changeTitle_Action(target.value));
  };

  return (
    <FlexRowDiv>
      <TextLabel htmlFor='controlTitle'>Title</TextLabel>
      <TextInput
        type='text'
        id='controlTitle'
        value={title}
        placeholder='Write Widget Title Here'
        onChange={(e) => handleInput(e)}
      />
    </FlexRowDiv>
  );
}
