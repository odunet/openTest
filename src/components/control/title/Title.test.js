import { render, screen } from '@testing-library/react';
import Title from './Title';
import toJson from 'enzyme-to-json';
import Enzyme, { shallow, mount } from 'enzyme';

describe('Render Title UI', () => {
  test('renders Title', () => {
    render(<Title />);
    const titleElement = screen.getByText(/Title/i);
    expect(titleElement).toBeInTheDocument();
  });
});

describe('Snapshot', () => {
  test('Should take a snapshot of Title component', () => {
    const container = shallow(<Title />);
    expect(toJson(container)).toMatchSnapshot();
  });
});
