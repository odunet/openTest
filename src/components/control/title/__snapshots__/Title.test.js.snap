// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Should take a snapshot of Title component 1`] = `
<Fragment>
  <styled.div>
    <styled.label
      htmlFor="controlTitle"
    >
      Title
    </styled.label>
    <styled.input
      id="controlTitle"
      onChange={[Function]}
      placeholder="Write Widget Title Here"
      type="text"
      value=""
    />
  </styled.div>
</Fragment>
`;
